crates/miden-tx/tests/integration/scripts/compatibility.rs

```diff
@@ -0,0 +1,490 @@
+
+    // TODO: implement
+}
+
+#[test]
+fn test_custom_account_custom_notes() {
+    // STATELESS CHECK
+    // --------------------------------------------------------------------------------------------
+
+    let account_custom_code_source = "
+        export.procedure_1
+            push.1.2.3.4 dropw
+        end
+
+        export.procedure_2
+            push.5.6.7.8 dropw
+        end
+    ";
+
+    let account_component = AccountComponent::compile_with_path(
+        account_custom_code_source,
+        TransactionKernel::testing_assembler(),
+        vec![],
+        LibraryPath::new("test::account::component_1").unwrap(),
+    )
+    .unwrap()
+    .with_supports_all_types();
+
+    let target_account = AccountBuilder::new(ChaCha20Rng::from_entropy().gen())
+        .with_component(account_component.clone())
+        .build_existing()
+        .unwrap();
+    let target_account_interface = AccountInterface::from(&target_account);
+
+    let mut mock_chain = MockChain::with_accounts(&[target_account.clone()]);
+    let sender_account =
+        mock_chain.add_existing_wallet(Auth::BasicAuth, vec![FungibleAsset::mock(20)]);
+    let serial_num =
+        RpoRandomCoin::new([Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)]).draw_word();
+    let tag = NoteTag::from_account_id(target_account.id(), NoteExecutionMode::Local).unwrap();
+    let metadata = NoteMetadata::new(
+        sender_account.id(),
+        NoteType::Public,
+        tag,
+        NoteExecutionHint::always(),
+        Default::default(),
+    )
+    .unwrap();
+    let vault = NoteAssets::new(vec![FungibleAsset::mock(100)]).unwrap();
+
+    let compatible_source_code = "
+        use.miden::contracts::wallets::basic->wallet
+        use.test::account::component_1->test_account
+
+        begin
+            push.1
+            if.true 
+                call.wallet::receive_asset
+                call.test_account::procedure_1
+            else
+                call.test_account::procedure_1
+                call.test_account::procedure_2
+            end
+        end
+    ";
+    let note_script = NoteScript::compile(
+        compatible_source_code,
+        TransactionKernel::testing_assembler()
+            .with_library(account_component.library())
+            .unwrap(),
+    )
+    .unwrap();
+    let recipient = NoteRecipient::new(serial_num, note_script, NoteInputs::default());
+    let compatible_custom_note = Note::new(vault.clone(), metadata, recipient);
+    assert_eq!(
+        CheckResult::Maybe,
+        target_account_interface.can_consume(&compatible_custom_note)
+    );
+
+    let incompatible_source_code = "
+        use.miden::contracts::wallets::basic->wallet
+        use.test::account::component_1->test_account
+
+        begin
+            push.1
+            if.true 
+                call.wallet::receive_asset
+                call.test_account::procedure_1
+            else
+                call.test_account::procedure_2
```

Comment:
We should be able to add `call.test_account::procedure_1` in the else branch without changing the test outcome, right?